homeassistant/components/climate/evohome.py

```diff
@@ -53,156 +65,420 @@
+class EvoClimateDevice(ClimateDevice):
+    """Base for a Honeywell evohome Climate device."""
+
+    # pylint: disable=no-member
+
+    def __init__(self, evo_data, client, obj_ref):
+        """Initialize the evohome entity."""
+        self._client = client
+        self._obj = obj_ref
 
-        self._config = evo_data['config'][GWS][0][TCS][0]
         self._params = evo_data['params']
         self._timers = evo_data['timers']
-
-        self._timers['statusUpdated'] = datetime.min
         self._status = {}
 
         self._available = False  # should become True after first update()
 
+    async def async_added_to_hass(self):
+        """Run when entity about to be added."""
+        async_dispatcher_connect(self.hass, DISPATCHER_EVOHOME, self._connect)
+
+    @callback
+    def _connect(self, packet):
+        if packet['to'] & self._type:
+            if packet['signal'] == 'refresh':
+                self.async_schedule_update_ha_state(force_refresh=True)
+
     def _handle_requests_exceptions(self, err):
-        # evohomeclient v2 api (>=0.2.7) exposes requests exceptions, incl.:
-        # - HTTP_BAD_REQUEST, is usually Bad user credentials
-        # - HTTP_TOO_MANY_REQUESTS, is api usuage limit exceeded
-        # - HTTP_SERVICE_UNAVAILABLE, is often Vendor's fault
+        # evohomeclient v2 api (>=0.2.7) now exposes requests exceptions, e.g.:
+        # - HTTP_BAD_REQUEST, is usually bad user credentials
+        # - HTTP_TOO_MANY_REQUESTS, is api usage limit exceeded
+        # - HTTP_SERVICE_UNAVAILABLE, is often vendor's fault
 
         if err.response.status_code == HTTP_TOO_MANY_REQUESTS:
-            # execute a back off: pause, and reduce rate
-            old_scan_interval = self._params[CONF_SCAN_INTERVAL]
-            new_scan_interval = min(old_scan_interval * 2, SCAN_INTERVAL_MAX)
-            self._params[CONF_SCAN_INTERVAL] = new_scan_interval
+            # execute a backoff: pause, and also reduce rate
+            old_interval = self._params[CONF_SCAN_INTERVAL]
+            new_interval = min(old_interval, SCAN_INTERVAL_DEFAULT) * 2
+            self._params[CONF_SCAN_INTERVAL] = new_interval
 
             _LOGGER.warning(
-                "API rate limit has been exceeded: increasing '%s' from %s to "
-                "%s seconds, and suspending polling for %s seconds.",
+                "API rate limit has been exceeded. Suspending polling for %s "
+                "seconds, and increasing '%s' from %s to %s seconds.",
+                new_interval * 3,
                 CONF_SCAN_INTERVAL,
-                old_scan_interval,
-                new_scan_interval,
-                new_scan_interval * 3
+                old_interval,
+                new_interval,
             )
 
             self._timers['statusUpdated'] = datetime.now() + \
-                timedelta(seconds=new_scan_interval * 3)
+                timedelta(seconds=new_interval * 3)
 
         else:
             raise err
```

Comment:
We don't `raise` normally. Log error and return.

Reply:
@MartinHjelmare I raise because I don't handle any other error. 

It's like asking me to trap all (say) `requests.exceptions.ConnectionError` just because it's possible they'll be thrown inside my `try`.  Should I then change the code 
```
except HTTPError as err:
```
to
```
except ConnectionError, HTTPError as err:
```

We had this discussion back in the initial release, and it was OK then... 

I'd like to leave it as is, OK?

Reply:
What I said back then, is that I'm not raising an exception, merely trapping some, and let the others through...

Reply:
@MartinHjelmare your thoughts on this?  As I said, I am not _creating_ a raise, merely trapping some & letting the others pass through.

I am not exactly sure of the "Integration Quality Scale", however, but I can remove the raise if you insist (I will put in a generic message), or can we go ahead with the PR as is?

Reply:
It sounds good to not catch things we don't expect.

Reply:
OK, resolved!